workflow/controller/operator.go

```diff
@@ -1226,16 +1226,20 @@ func (woc *wfOperationCtx) assessNodeStatus(pod *apiv1.Pod, node *wfv1.NodeStatu
 		// we only need to update these values if the container transitions to complete
 		if newPhase.Fulfilled() {
 			// outputs are mixed between the annotation (parameters, artifacts, and result) and the pod's status (exit code)
+			outputs := &wfv1.Outputs{}
```

Comment:
does this need changing?

Reply:
Yes,
In deadlineExceed scenario,
it won't have an exit code, thus won't annotate the output in the original code
this fixed that